MovieApp/PresentationLayer/HomeScreen/CategorisedCollection/CategorisedCollectionView.swift

```diff
@@ -79,11 +80,18 @@ extension CategorisedCollectionView: UICollectionViewDataSource {
                 withReuseIdentifier: MoviePosterCell.cellIdentifier,
                 for: indexPath) as? MoviePosterCell
         else {
-            return MoviePosterCell()
+            let cell = MoviePosterCell()
+            return cell
         }
 
         let movie = movies[indexPath.row]
         cell.setData(for: movie)
+        cell.moviePoster.onFavoriteToggle = { [self] movieId in
+            movies = movies.map {
```

Comment:
Do not update UI from here, call presenter to update the flag and then force the data reload.